### PR TITLE
use more inclusive terms in the code

### PR DESCRIPTION
**Pull request by shintaro-iwasaki**
## Pull Request Description

This patch makes a language used in the MPICH code more inclusive. This change is not complete because of the dependency on branch names of some repositories, which will be solved soon.

[EDIT] The changes made by this PR is as follows:
#### configure
`master_top_srcdir` -> `main_top_srcdir`

#### git-related
"master" -> "main" branch (although the actual branch name has not been changed yet)

#### examples/pmandel*
slave->child
master->parent

#### global structures and variables
`MPIR_ThreadInfo.master_thread` -> `MPIR_ThreadInfo.main_thread`
`struct PMIMaster` -> `struct PMIMain`
`PMI2_Connect_comm_t.isMaster` -> `PMI2_Connect_comm_t.isMain`

#### test names
`spawntest_master` -> `spawntest_parent`
`taskmaster` -> `taskmanager`
`th_taskmaster` -> th_taskmanager`

#### comments and names of temporary/local variables:
master -> main, parent, server, ...
slave -> child, worker, client, ...
This change depends on the context.

## Expected Impact

This PR does not and should not change any functionality of MPICH.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [x] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] You or your company has a signed contributor's agreement on file with Argonne
* [x] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Comment by shintaro-iwasaki** [comment]
test:mpich/ch4/most
test:mpich/ch3/most

**Comment by hzhou** [comment]
@shintaro-iwasaki Could you describe/list the replacement in the PR description and commit messages?

**Comment by shintaro-iwasaki** [comment]
@hzhou Thanks! I updated the commit message.
test:mpich/ch4/most

**Comment by shintaro-iwasaki** [comment]
test:mpich/ch3/most

**Comment by shintaro-iwasaki** [comment]
All the ch3,ch4/most tests have passed. I will rebase it and merge this PR.

**Comment by shintaro-iwasaki** [comment]
Pushing the current master branch closed this PR. I am reopening it.

File: CONTRIBUTING [hunk 1]
```diff
@@ -7,7 +7,7 @@ agreement. http://www.mpich.org/documentation/contributor-docs/
 standards. https://wiki.mpich.org/mpich/index.php/Coding_Standards
 
 3. Use the git pre-commit hook included with MPICH
-(https://github.com/pmodels/mpich/blob/master/maint/hooks/pre-commit)
+(mpich_root/maint/hooks/pre-commit)
```
Comment (wesbland) [inline CONTRIBUTING:10]: I understand this change, but if you guys are planning to change the branch name to `main`, I’d like to see the link put back.

**Pull request merged**